src/display_methods.jl

```diff
@@ -31,9 +31,17 @@ const default_mime_types = ["image/svg+xml", "image/png", "text/html", "text/pla
 #From IJulia as a reminder
 #const supported_mime_types = [ "text/html", "text/latex", "image/svg+xml", "image/png", "image/jpeg", "text/plain", "text/markdown" ]
 
+const mimetype_from_fig_ext =
+    Dict(".png" => "image/png",
+         ".jpg" => "image/jpeg",
+         ".jpeg" => "image/jpeg",
+         ".svg" => "image/svg+xml",
+         ".pdf" => "application/pdf")
+
 function Base.display(report::Report, data)
     #Set preferred mimetypes for report based on format
-    for m in report.mimetypes
+    fig_ext = report.cur_chunk.options[:fig_ext]
+    for m in Iterators.flatten(((mimetype_from_fig_ext[fig_ext], ), report.mimetypes))
```

Comment:
I think you should call `unique` here, otherwise you can try to display the same mimetype twice.

Reply:
I think `unique([mimetype_from_fig_ext[fig_ext]; report.mimetypes])` would be more readable here instead of flatten. I can also change it if you want.

Reply:
Yes please make any changes you think are appropriate...I'm pretty swamped at the moment.

Reply:
OK